x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/QuerySettings.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.logical;
+
+import org.elasticsearch.xpack.esql.core.expression.Alias;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+
+import java.util.List;
+import java.util.Objects;
+
+public class QuerySettings {
+
+    private final List<Alias> fields;
```

Comment:
I wonder if it is worth storing it in a `name->value` map? I suspect it will be accessed by name most of the times